src/lib/api-transport.ts

```diff
@@ -0,0 +1,91 @@
+import Transport from 'winston-transport';
+import { Socket } from 'socket.io-client';
+
+export type ApiTransportSettings = {
+	sendingEnabled?: boolean;
+	bufferSize?: number;
+	sendInterval?: number;
+};
+
+export type ApiTransportOptions = Transport.TransportStreamOptions & ApiTransportSettings & { socket?: Socket };
+
+type Info = {
+	message: string;
+	timestamp: string;
+	level: string;
+	scope: string;
+};
+
+class ApiTransport extends Transport {
+	public socket: Socket | undefined;
+	private sendingEnabled: boolean;
+	private bufferSize: number;
+	private sendInterval: number;
+	private logBuffer: Info[] = [];
+	private droppedLogs: number = 0;
+	private timer: NodeJS.Timeout | undefined = undefined;
+
+	constructor (opts?: ApiTransportOptions) {
+		super(opts);
+		this.sendingEnabled = opts?.sendingEnabled ?? false;
+		this.bufferSize = opts?.bufferSize ?? 100;
+		this.sendInterval = opts?.sendInterval ?? 10000;
+		this.socket = opts?.socket;
+		this._setInterval();
+	}
+
+	override log (info: Info, callback?: () => void) {
+		setImmediate(() => this.emit('logged', info));
+
+		this.logBuffer.push(info);
+		const bufferLength = this.logBuffer.length;
+		const bufferOverflow = bufferLength - this.bufferSize;
+
+		if (bufferOverflow > 0) {
+			this.logBuffer = this.logBuffer.slice(bufferOverflow);
+			this.droppedLogs += bufferOverflow;
+		}
+
+		callback && callback();
+	}
+
+	_setInterval () {
+		this.timer && clearInterval(this.timer);
+
+		if (this.sendingEnabled) {
+			this.timer = setInterval(() => this._sendLogs(), this.sendInterval);
+		}
+	}
+
+	_sendLogs () {
+		if (!this.sendingEnabled || !this.socket?.connected || !this.logBuffer.length) {
+			return;
+		}
+
+		const payload = {
+			logs: this.logBuffer,
+			skipped: this.droppedLogs,
+		};
+
+		this.socket.emit('probe:logs', payload);
+		this.logBuffer = [];
+		this.droppedLogs = 0;
```

Comment:
We need to get [ack](https://socket.io/docs/v3/emitting-events/#acknowledgements) from server before clearing the buffer, so in case of e.g. `Probe not found` error we are retrying to send prev log messages again.